src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,198 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * The Storage class deals with loading tasks from the file and saving tasks in the file.
+ *
+ * @author Inez Kok
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * This is the constructor for a Storage.
+     *
+     * @param filePath The string representation of the file path.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Checks whether the string representation of a date is valid.
+     *
+     * @param date The string representation of the date.
+     * @throws DukeException On format error.
+     */
+    private void validateDate(String date) throws DukeException {
```

Comment:
would it be better if the method returned true or false instead of throwing an exception?

Reply:
Actually nvrmind, I can see why you are throwing the duke exception.